browser/brave_shields/brave_shields_tab_helper.cc

```diff
@@ -53,7 +54,13 @@ BraveShieldsTabHelper::BraveShieldsTabHelper(content::WebContents* web_contents)
       content::WebContentsUserData<BraveShieldsTabHelper>(*web_contents) {
   favicon::ContentFaviconDriver::FromWebContents(web_contents)
       ->AddObserver(this);
-  observation_.Observe(GetHostContentSettingsMap(web_contents));
+  auto* map = GetHostContentSettingsMap(web_contents);
+  observation_.Observe(map);
+  PrefService* profile_prefs =
+      Profile::FromBrowserContext(web_contents->GetBrowserContext())
+          ->GetPrefs();
+  brave_shields_settings_ = new BraveShieldsSettings(
```

Comment:
pass in `BraveShieldsSettings` and `HostContentSettingsMap` to the constructor. This doc needs more internal links, but see dependency injection/inversion https://chromium.googlesource.com/chromium/src/+/main/docs/chrome_browser_design_principles.md